Zad02

sm000k opened the pull request.

sm000k committed:
Zad02

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -0,0 +1,5 @@
+package pl.zimi.zadanka;
+
+public class Zad2 {
```
ojejku jak nieładnie

sm000k committed:
Zad02

sm000k committed:
Zad02 to jest moj kod w intelijj, ale nie wiem zrobić solution

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
+    }
+
+    //System.out.println
+    static Integer findMissingNumber(List<Integer> l) {
```
zmień tą linię na:
public int missingNumber(int[] nums) {

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
+    public static void main(String[] args) {
+        List<Integer> inputNumbers = new ArrayList<>();
+        inputNumbers = getInput();
+        Integer x = findMissingNumber(inputNumbers);
```
new Solution().missingNumber(inputNumbers.toArray(new Integer[inputNumbers.size()])));

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
+        for (String s : splicedString) {
+            inputNumbers.add(Integer.parseInt(s));
+        }
+        Collections.sort(inputNumbers);
```
to powinno być w metodzie missingNumber

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
+        Integer i = 0;
+        Integer k = 0;
+
+        if (l.get(0) == k) {
```
use equals method

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
+            while (l.get(i).equals(i)) {
+                i++;
+
+                if (i.intValue() == l.size()) {
```
use equals

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
 public class Zad2 {
+    static List<Integer> getInput() {
+        List<Integer> inputNumbers = new ArrayList<>();
+        String inputLine;
```
połącz deklaracje z przypisaniem

czyli odrazu
String inputLine = scanner.next();

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
+    static List<Integer> getInput() {
+        List<Integer> inputNumbers = new ArrayList<>();
+        String inputLine;
+        String inputLineNoBrackets;
```
połącz deklarację z przypisaniem

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -1,8 +1,68 @@
+    ;
+
+    public static void main(String[] args) {
+        List<Integer> inputNumbers = new ArrayList<>();
+        inputNumbers = getInput();
```
połącz w jedną linijkę

sm000k committed:
Zad02

Linia 63 wywala błąd :
Cannot resolve method 'missingNumber' in 'Solution'

nie wiem czy poprawnie dodałem klasę Solution

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -23,46 +23,46 @@
-
     public static void main(String[] args) {
         List<Integer> inputNumbers = new ArrayList<>();
         inputNumbers = getInput();
```
połącz w jedną metodę

zimi-pl commented on src/main/java/pl/zimi/zadanka/Zad2.java:
```diff
@@ -23,46 +23,46 @@
-        System.out.println(x);
-
+        //        Integer x = findMissingNumber(inputNumbers);
+        System.out.println(new Solution().missingNumber(inputNumbers.toArray(new Integer[inputNumbers.size()]))););
```
Solution to w założeniu nazwa klasy
w leet code oczekiwana nazwa klasy to Solution
dlatego ja dałem Solution
Ale Twoja klasa nazywa się Zad2

żeby działało i tu i tu musisz zmienić nazwę klasy i pliku na Solution i wtedy ta linia może zostać i będzie działać

sm000k replied:
Zgadza się podmianka nazwy klasy pozwala uruchomić kod

sm000k committed:
Zad02

Linia 63 wywala błąd :
nazwa klasy głownej musi być zbieżna z nazwą projektu

zimi-pl committed:
jsoup, @Test

sm000k committed:
Usunieto test bledny

zimi-pl commented:
super
